src/x_config_gen/device_section.rs

```diff
@@ -0,0 +1,207 @@
+            DeviceSectionElement::BusId(value) => writeln!(f, "BusID \"{}\"", value),
+            DeviceSectionElement::Irq(value) => writeln!(f, "IRQ {}", value),
+            DeviceSectionElement::Screen(value) => writeln!(f, "IRQ {}", value),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Default)]
+pub struct DeviceSection {
+    content: Vec<DeviceSectionElement>,
+    has_identifier: bool,
+    has_vendor: bool,
+    has_board: bool,
+    has_chipset: bool,
+    has_card: bool,
+    has_driver: bool,
+    has_ramdac: bool,
+    has_dacspeed: bool,
+    has_videoram: bool,
+    has_membase: bool,
+    has_iobase: bool,
+    has_clockchip: bool,
+    has_chipid: bool,
+    has_chiprev: bool,
+    has_clocks: bool,
+    has_matchseat: bool,
+    has_busid: bool,
+    has_irq: bool,
+    has_screen: bool,
+}
+
+#[allow(dead_code)]
+impl DeviceSection {
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    create_function!(add_comment, &str, DeviceSectionElement::Comment);
+    create_function!(
+        set_identifier,
+        &str,
+        DeviceSectionElement::Identifier,
+        has_identifier
+    );
+    create_function!(set_vendor, &str, DeviceSectionElement::Vendor, has_vendor);
+    create_function!(set_board, &str, DeviceSectionElement::Board, has_board);
+    create_function!(
+        set_chipset,
+        &str,
+        DeviceSectionElement::Chipset,
+        has_chipset
+    );
+    create_function!(set_card, &str, DeviceSectionElement::Card, has_card);
+    create_function!(set_driver, &str, DeviceSectionElement::Driver, has_driver);
+    create_function!(set_ramdac, &str, DeviceSectionElement::RamDac, has_ramdac);
+    create_function!(
+        set_dacspeed,
+        &Vec<i32>,
+        DeviceSectionElement::DacSpeed,
+        has_dacspeed
+    );
+    create_function!(
+        set_videoram,
+        i32,
+        DeviceSectionElement::VideoRam,
+        has_videoram
+    );
+    create_function!(set_membase, u64, DeviceSectionElement::MemBase, has_membase);
+    create_function!(set_iobase, u64, DeviceSectionElement::IoBase, has_iobase);
+    create_function!(
+        set_clockship,
+        &str,
+        DeviceSectionElement::ClockChip,
+        has_clockchip
+    );
+    create_function!(set_chipid, i32, DeviceSectionElement::ChipId, has_chipid);
+    create_function!(set_chiprev, i32, DeviceSectionElement::ChipRev, has_chiprev);
+    create_function!(
+        set_clocks,
+        &Vec<i32>,
+        DeviceSectionElement::Clocks,
+        has_clocks
+    );
+    create_function!(
+        set_matchset,
+        &str,
+        DeviceSectionElement::MatchSeat,
+        has_matchseat
+    );
+    create_function!(set_busid, &str, DeviceSectionElement::BusId, has_busid);
+    create_function!(set_irq, i32, DeviceSectionElement::Irq, has_irq);
+    create_function!(set_screen, i32, DeviceSectionElement::Screen, has_screen);
```

Comment:
not sure how much you care about api idiomacy, but this looks like you want to use a builder pattern